src/NuGet.Core/NuGet.Build.Tasks/NuGet.Build.Tasks.Pack.targets

```diff
@@ -209,7 +210,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </Target>
 
   <Target Name="GenerateNuspec"
-          Condition="'$(IsPackable)' == 'true'"
+          Condition="'$(IsPackable)' == 'true' AND '$(PackageReferenceCompatibleProjectStyle)' == 'true'"
```

Comment:
> Btw, we still need to make sure pack doesn't run on packages.config projects, so idk if the issue linked can actually be closed.

@nkolev92 I believe the change to this target's condition achieves this goal.

Reply:
weird, I did not see that, you're right.